src/error.ts

```diff
@@ -0,0 +1,53 @@
+import { STATUS_CODES } from 'node:http';
+import { format } from 'node:util';
+
+/**
+ * Base HTTP error class built with a message and a status code
+ * @extends Error
+ */
+export class ErrorHTTP extends Error {
+  status: number;
+
+  constructor(message?: string | number, status: number = 500) {
+    super();
+
+    if (!message) {
+      this.status = status;
+      this.message = STATUS_CODES[status] as string;
+    } else if (typeof message === 'number') {
+      this.status = message;
+      this.message = STATUS_CODES[this.status] as string;
```

Comment:
nit: seems a bit confusing from a user's perspective that the message passed in can become the status code if it's a number. Wonder if it makes more sense to just allow message to only be a string and always set this.status to status passed in.

Reply:
This is a confusing API from Node.js core IMO. The STATUS_CODES structure looks like this (abridged):

```js
const STATUS_CODES = {
  '404': 'Not Found',
  404: 'Not Found'
}
```

So the intent here is to support a call like `new ErrorHTTP(404)` and get the default HTTP message from node core, such as "Not Found". It isn't turning the 404 into a string, per se. 

The type mutation `as string` was a lazy way of getting around a type error "Type 'string | undefined' is not assignable to type 'string'." but in hindsight it's better to fall back to a string like "Unknown status". I've done this in 05a68d9 - WDYT?

Reply:
Yeah that makes sense to me, thanks for making that change!